flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/TimeBoundedStreamJoin.scala

```diff
@@ -382,6 +498,29 @@ abstract class TimeBoundedStreamInnerJoin(
     }
   }
 
+  /**
+    * Return a padding result with the given left/right row.
+    * @param row the row to pad
+    * @param paddingLeft pad left or right
+    * @return the null padding result
+    */
+  private def paddingResult(row: Row, paddingLeft: Boolean): Row = {
```

Comment:
Move this method into a util class and split it into two method (`padLeft` and `padRight`). The methods should be `final`.